cast that performs the conversions of a reinterpret_cast is not allowed in a constant expression

I've come across this error on clang-11:

```
../libs/kdmp-parser/src/lib/kdmp-parser-structs.h:39:20: error: constexpr function never produces a constant expression [-Winvalid-constexpr]
constexpr uint64_t OffsetFromThis(const void *This, const void *Field) {
                   ^
../libs/kdmp-parser/src/lib/kdmp-parser-structs.h:40:10: note: cast that performs the conversions of a reinterpret_cast is not allowed in a constant expression
  return uint64_t(Field) - uint64_t(This);
         ^
1 error generated.
```

--- a/.github/workflows/kdmp-parser.yml
+++ b/.github/workflows/kdmp-parser.yml
@@ -32,14 +32,14 @@ jobs:
         sudo bash -c "$(wget -O - https://apt.llvm.org/llvm.sh)"
 
     - name: Build with gcc
-      if: $${ matrix.compiler } == 'gcc'
+      if: ${{ matrix.compiler }} == 'gcc'
       run: python3 builder.py --run-tests
       env:
         CC: gcc
         CXX: g++
 
     - name: Build with clang
-      if: $${ matrix.compiler } == 'clang'
+      if: ${{ matrix.compiler }} == 'clang'
       run: python3 builder.py --run-tests
       env:
         CC: clang

--- a/.github/workflows/kdmp-parser.yml
+++ b/.github/workflows/kdmp-parser.yml
@@ -13,7 +13,7 @@ jobs:
         pyver: ['3.6', '3.7', '3.8', '3.9', '3.x']
         compiler: ['clang', 'gcc']
 
-    name: Ubuntu 20.04 - Py${{ matrix.pyver }} / ${{ matrix.cxx }}
+    name: Ubuntu 20.04 - Py${{ matrix.pyver }} / ${{ matrix.compiler }}
     runs-on: ubuntu-20.04
     steps:
     - name: Checkout

--- a/.github/workflows/kdmp-parser.yml
+++ b/.github/workflows/kdmp-parser.yml
@@ -18,7 +18,7 @@ jobs:
       CC: ${{ matrix.cc }}
       CXX: ${{ matrix.cxx }}
 
-    name: Ubuntu 20.04 - Py${{ matrix.pyver }}
+    name: Ubuntu 20.04 - Py${{ matrix.pyver }} / ${{ matrix.cxx }}
     runs-on: ubuntu-20.04
     steps:
     - name: Checkout
@@ -37,7 +37,7 @@ jobs:
         sudo bash -c "$(wget -O - https://apt.llvm.org/llvm.sh)"
 
     - name: Build
-      run: python3 builder.py --run-tests
+      run: CC=${{ matrix.cc }} CXX=${{ matrix.cxx }} python3 builder.py --run-tests
 
     - name: Upload artifacts for rel/x64
       uses: actions/upload-artifact@v2

--- a/.github/workflows/kdmp-parser.yml
+++ b/.github/workflows/kdmp-parser.yml
@@ -32,18 +32,18 @@ jobs:
         sudo bash -c "$(wget -O - https://apt.llvm.org/llvm.sh)"
 
     - name: Build with gcc
-      if: ${{ matrix.compiler }} == 'gcc'
+      if: matrix.compiler == 'gcc'
       run: python3 builder.py --run-tests
       env:
-        CC: gcc
-        CXX: g++
+        CC: gcc-10
+        CXX: g++-10
 
     - name: Build with clang
-      if: ${{ matrix.compiler }} == 'clang'
+      if: matrix.compiler == 'clang'
       run: python3 builder.py --run-tests
       env:
-        CC: clang
-        CXX: clang++
+        CC: clang-11
+        CXX: clang++-11
 
     - name: Upload artifacts for rel/x64
       uses: actions/upload-artifact@v2

--- a/.github/workflows/kdmp-parser.yml
+++ b/.github/workflows/kdmp-parser.yml
@@ -4,13 +4,19 @@ name: Builds
 on: [push, pull_request]
 
 jobs:
-  Linux2004Gcc:
+  Ubuntu2004:
     strategy:
       fail-fast: false
       matrix:
         # Available versions: https://raw.githubusercontent.com/actions/python-versions/main/versions-manifest.json
         # We don't test below 3.6 because f'strings were implemented in 3.6 and we use them in build scripts, etc.
         pyver: ['3.6', '3.7', '3.8', '3.9', '3.x']
+        cc: ['clang-11', 'gcc-10']
+        cxx: ['clang++-11', 'g++-10']
+
+    env:
+      CC: ${{ matrix.cc }}
+      CXX: ${{ matrix.cxx }}
 
     name: Ubuntu 20.04 - Py${{ matrix.pyver }}
     runs-on: ubuntu-20.04
@@ -27,7 +33,8 @@ jobs:
     - name: Install dependencies
       run: |
         sudo apt-get -y update
-        sudo apt install -y g++-multilib ninja-build
+        sudo apt install -y g++-multilib ninja-build g++-10
+        sudo bash -c "$(wget -O - https://apt.llvm.org/llvm.sh)"
 
     - name: Build
       run: python3 builder.py --run-tests

--- a/.github/workflows/kdmp-parser.yml
+++ b/.github/workflows/kdmp-parser.yml
@@ -28,7 +28,7 @@ jobs:
     - name: Install dependencies
       run: |
         sudo apt-get -y update
-        sudo apt install -y g++-multilib ninja-build g++-10
+        sudo apt install -y g++-10-multilib g++-10 ninja-build
         sudo bash -c "$(wget -O - https://apt.llvm.org/llvm.sh)"
 
     - name: Build with gcc

--- a/.github/workflows/kdmp-parser.yml
+++ b/.github/workflows/kdmp-parser.yml
@@ -11,12 +11,7 @@ jobs:
         # Available versions: https://raw.githubusercontent.com/actions/python-versions/main/versions-manifest.json
         # We don't test below 3.6 because f'strings were implemented in 3.6 and we use them in build scripts, etc.
         pyver: ['3.6', '3.7', '3.8', '3.9', '3.x']
-        cc: ['clang-11', 'gcc-10']
-        cxx: ['clang++-11', 'g++-10']
-
-    env:
-      CC: ${{ matrix.cc }}
-      CXX: ${{ matrix.cxx }}
+        compiler: ['clang', 'gcc']
 
     name: Ubuntu 20.04 - Py${{ matrix.pyver }} / ${{ matrix.cxx }}
     runs-on: ubuntu-20.04
@@ -36,8 +31,19 @@ jobs:
         sudo apt install -y g++-multilib ninja-build g++-10
         sudo bash -c "$(wget -O - https://apt.llvm.org/llvm.sh)"
 
-    - name: Build
-      run: CC=${{ matrix.cc }} CXX=${{ matrix.cxx }} python3 builder.py --run-tests
+    - name: Build with gcc
+      if: $${ matrix.compiler } == 'gcc'
+      run: python3 builder.py --run-tests
+      env:
+        CC: gcc
+        CXX: g++
+
+    - name: Build with clang
+      if: $${ matrix.compiler } == 'clang'
+      run: python3 builder.py --run-tests
+      env:
+        CC: clang
+        CXX: clang++
 
     - name: Upload artifacts for rel/x64
       uses: actions/upload-artifact@v2

--- a/src/lib/kdmp-parser-structs.h
+++ b/src/lib/kdmp-parser-structs.h
@@ -36,14 +36,14 @@ enum class DumpType_t : uint32_t { FullDump = 1, KernelDump = 2, BMPDump = 5 };
 // returns the offset of Field in the object via pointer arithmetic.
 //
 
-constexpr uint64_t OffsetFromThis(const void *This, const void *Field) {
+constexpr uint64_t OffsetFromThis(const uintptr_t This, const uintptr_t Field) {
   return uint64_t(Field) - uint64_t(This);
 }
 
 static void DisplayHeader(const uint32_t Prefix, const char *FieldName,
                           const void *This, const void *Field) {
-  printf("%*s+0x%04" PRIx64 ": %-25s", Prefix, "", OffsetFromThis(This, Field),
-         FieldName);
+  printf("%*s+0x%04" PRIx64 ": %-25s", Prefix, "",
+         OffsetFromThis(uintptr_t(This), uintptr_t(Field)), FieldName);
 }
 
 //

